From 355b31a8f4e1cdf09038c50c08f67c1c09093225 Mon Sep 17 00:00:00 2001
From: Xiao,Feng <fengx.xiao@intel.com>
Date: Thu, 10 Jan 2013 11:07:27 +0800
Subject: AVIExtractor: Provide video framerate info

BZ: 46801

For the request form HDMI dynamic setting

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: no

Change-Id: I02c4e82249efee0fe361ace33e358a4c458d4ace
Orig-Change-Id: I9138bf7e0441bc4a6e8c2900b569fab3825d0ff7
Signed-off-by: Xiao,Feng <fengx.xiao@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index da4fe8e..563e8e6 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -935,6 +935,13 @@ status_t AVIExtractor::parseIndex(off64_t offset, size_t size) {
         AString mime = tmp;
 
         if (!strncasecmp("video/", mime.c_str(), 6)) {
+            if (durationUs != 0) {
+                int32_t frameRate = (track->mSamples.size() * 1000000LL + (durationUs >> 1)) / durationUs;
+                if (frameRate > 0) {
+                    track->mMeta->setInt32(kKeyFrameRate, frameRate);
+                    ALOGV("video framerate = %d", frameRate);
+                }
+            }
             if (track->mThumbnailSampleIndex >= 0) {
                 int64_t thumbnailTimeUs;
                 CHECK_EQ((status_t)OK,
-- 
1.7.4.1

